pkg/model/requests_processing.go

```diff
@@ -56,21 +58,37 @@ func (rps *requestsProcessingStock) EntitiesInStock() []*simulator.Entity {
 }
 
 func (rps *requestsProcessingStock) Remove() simulator.Entity {
-	return rps.delegate.Remove()
+	request := rps.delegate.Remove().(*requestEntity)
+	*rps.currentUtilization -= request.requestConfig.CPUUtilization
+	return request
 }
 
 func (rps *requestsProcessingStock) Add(entity simulator.Entity) error {
 	rps.numRequestsSinceLast++
+	request := *entity.(*requestEntity)
+	*rps.currentUtilization += request.requestConfig.CPUUtilization
 
 	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
-	totalTime := calculateTime(rps.delegate.Count(), rps.replicaMaxRPSCapacity, time.Second, rng)
-
-	rps.env.AddToSchedule(simulator.NewMovement(
-		"complete_request",
-		rps.env.CurrentMovementTime().Add(totalTime),
-		rps,
-		rps.requestsComplete,
-	))
+	totalTime := calculateTime(*rps.currentUtilization, *rps.totalCPUCapacity, time.Second, rng)
+	totalTime += time.Duration(request.requestConfig.IOUtilization)
+
+	if totalTime > request.requestConfig.Timeout {
+		*rps.currentUtilization -= request.requestConfig.CPUUtilization
```

Comment:
This is probably fine for now.  But as we refine the model we might want to include CPU usage for the duration of the timeout.

Reply:
I got your idea. It really matters to tune in the model. I designed a new model (see e91267565b8ba92593c89fe969f1d65a35e99baf commit). For now I didn't tune this part. Definitely I'll fix it in next refining.